### PR TITLE
[WIP] Add new caching service

### PR DESCRIPTION
**Pull request by Plootie**
Currently just putting this up here so people can give feedback before I commit to this design

### CODE REVIEW

**Comment by Plootie** [comment]
I've probably messed something up massively but I think I've gotten this to a place where I'm happy to get some feedback

File: SIT.Manager/ViewModels/PlayPageViewModel.cs [hunk 1]
```diff
@@ -87,6 +90,15 @@ public partial class PlayPageViewModel : ObservableObject
...
         QuickPlayCommand = new AsyncRelayCommand(async () => await ConnectToServer(true));
+
+        cachingService.InMemory.Add("testing", "uwu", TimeSpan.FromSeconds(4));
+        for (int i = 0; i < 3; i++)
+        {
+            Thread.Sleep(1750);
+            Debug.WriteLine($"Interation {i}");
+            Debug.WriteLine(cachingService.InMemory.Get<string>("testing"));
+            Debug.WriteLine("");
+        }
```
Comment (artehe) [inline SIT.Manager/ViewModels/PlayPageViewModel.cs:101]: probably worth removing this before merging it

File: SIT.Manager/Services/Caching/OnDiskCachingProvider.cs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+            if (tType == typeof(string))
+            {
+                return new CacheValue<T>((T) (object) Encoding.UTF8.GetString(fileBytes), true);
+            }
+            return new CacheValue<T>(JsonSerializer.Deserialize<T>(fileBytes), true);
+        }
+        catch (Exception ex)
+        {
+            //TODO: log exception
```
Comment (artehe) [inline SIT.Manager/Services/Caching/OnDiskCachingProvider.cs:146]: Handle this todo please

File: SIT.Manager/Services/Caching/OnDiskCachingProvider.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                if (tType == typeof(byte[]))
+                {
+                    return new CacheValue<T>((T) (object) fileBytes, true);
+                }
+            }
+            catch (Exception ex)
+            {
+                ms.Dispose();
+                //TODO: logging
+                return CacheValue<T>.Null;
```
Comment (artehe) [inline SIT.Manager/Services/Caching/OnDiskCachingProvider.cs:135]: Handle this todo please

File: SIT.Manager/Services/Caching/InMemoryCachingProvider.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        try
+        {
+            T value = cacheEntry.GetValue<T>();
+            return new CacheValue<T>(value, true);
+        }
+        catch (Exception ex)
+        {
+            //TODO: log exception here
+            return CacheValue<T>.NoValue;
```
Comment (artehe) [inline SIT.Manager/Services/Caching/InMemoryCachingProvider.cs:38]: Handle this todo please

**Review by artehe — COMMENTED** [review]
I think it's worth handling the logging todos just so we have some idea if something breaks there and just remove the testing code in the play page (even though it will go in pr #38 anyway)

**Pull request merged**